modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/RoleResourceManager.java

```diff
@@ -369,6 +378,11 @@ public SCIMResponse updateWithPATCHRole(String id, String patchRequest, RoleMana
             SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getRoleResourceSchema();
             Map<String, Boolean> requestAttributes = ResourceManagerUtil.getAllAttributeURIs(schema);
 
+            List<PatchOperation> opList = getDecoder().decodeRequest(patchRequest);
```

Comment:
Any chance of getting null for opList? If so handle the possible NPE

Reply:
Fixed